aten/src/ATen/native/layer_norm.cpp

```diff
@@ -266,6 +269,28 @@ Tensor rms_norm_symint(
   const Tensor& weight = *weight_maybe_owned;
   _check_rms_norm_inputs_symint(input, normalized_shape, weight);
 
+#ifdef USE_MPS
+  if (weight_opt.has_value()) {
```

Comment:
This one would need a device check, wouldn't it? Otherwise it will try to use this path for CPU tensors...